src/execution/index/fixed_size_buffer.cpp

```diff
@@ -5,31 +5,65 @@
 
 namespace duckdb {
 
+//===--------------------------------------------------------------------===//
+// PartialBlockForIndex
+//===--------------------------------------------------------------------===//
+
+void PartialBlockForIndex::AddUninitializedRegion(idx_t start, idx_t end) {
+	throw InternalException("no uninitialized regions for PartialBlockForIndex");
```

Comment:
We should mark regions as uninitialized as discussed - could you add an index test to the storage initialization verification?